Bryanna/user profile page

bry-gavino opened the pull request:
Pages for Users to look at and edit their password and email.
(I added a "Users" link at the top of Layout just so it's easy to access it for now!)

## Screenshots

<img width="1680" alt="Screen Shot 2020-11-04 at 11 17 29 PM" src="https://user-images.githubusercontent.com/60284767/98209640-1fd77700-1ef4-11eb-9386-a116901c7fb9.png">
<img width="1675" alt="Screen Shot 2020-11-04 at 11 17 36 PM" src="https://user-images.githubusercontent.com/60284767/98209644-20700d80-1ef4-11eb-993a-f89d540a452b.png">


_Required for visual changes_

CC: @kwfk

bry-gavino committed:
docs: start

bry-gavino committed:
feat: new page for profile

bry-gavino committed:
feat: staging files

bry-gavino committed:
feat: git pulling

bry-gavino committed:
feat: user settings save and edit

kwfk commented on .env.example:
```diff
@@ -1,4 +1,5 @@
-DATABASE_URL="postgresql://postgres@localhost:5432/nbjc_development?schema=public"
+PG_USERNAME=postgres
+PG_PASSWORD=password
+PG_PORT=5432
```
this should match main

kwfk commented on pages/users/sample.tsx:
```diff
@@ -0,0 +1,7 @@
+  Password: 'peepeepoo',
+};
+
+export default SampleUser;
```
put this in `utils/sample_data.ts`

kwfk commented on pages/users/settings_edit.tsx:
```diff
@@ -0,0 +1,22 @@
+import SampleUser from './sample';
+import SettingsEdit from '../../components/user/edit';
+
+const UserProfSettings: FunctionComponent<{ initial?: number }> = () => {
```
use React.FunctionComponent or React.FC

bry-gavino committed:
feat: made one page that switches between components

bry-gavino committed:
feat: git pulled

kwfk commented on styles/users/SettingsEdit.module.css:
```diff
@@ -0,0 +1,55 @@
+.content {
```
separate out the stylesheet between components and the page
(take a look at the other components to see how they're structured)

kwfk replied:
if the multiple components have the same styles you can import the same stylesheet to multiple components
feel free to ask me for clarification

kwfk commented on pages/users/settings.tsx:
```diff
@@ -0,0 +1,86 @@
+import SettingsEdit from '../../components/user/edit';
+import SampleUser from './sample';
+
+const UserProfSettings: FunctionComponent<{ initial?: string }> = ({
+  initial = 'edit',
+}) => {
+  const [setting, setSetting] = useState(initial);
```
initial doesn't need to be a prop
you can initialize state by putting in the `useState` like `useState('edit')`

kwfk replied:
it might be good to actually use integers instead of string

kwfk commented on pages/users/settings.tsx:
```diff
@@ -0,0 +1,86 @@
+  initial = 'edit',
+}) => {
+  const [setting, setSetting] = useState(initial);
+  const hiddenPassword: string = '*'.repeat(SampleUser.Password.length);
```
let's use a constant length like 6

kwfk replied:
so just `hiddenPassword = '******'`

kwfk commented on pages/users/settings.tsx:
```diff
@@ -0,0 +1,86 @@
+
+          <SettingsEdit
+            userType={SampleUser.ProfileType}
+            email={SampleUser.Email}
+            password={hiddenPassword}
+          />
+
+          <div className={styles.delete}>
+            <Link>Delete User Account</Link>
+          </div>
+        </div>
+      );
+    }
+    return (
+      <div className={styles.box}>
+        <div className={styles.top}>
+          <div className={styles.title}>{SampleUser.ProfileType} Profile</div>
+          <div className={styles.settings}>User Account Settings</div>
+        </div>
+        <SettingsSave
+          userType={SampleUser.ProfileType}
+          email={SampleUser.Email}
+          password={SampleUser.Password}
+        />
+        <div className={styles.delete}>
+          <div>
+            <Link>Delete User Account</Link>
+          </div>
+          <div>
+            <Button
+              onClick={() => setSetting('edit')}
+              variant="contained"
+              color="primary"
+              disableElevation
+            >
+              Save
+            </Button>
+          </div>
+        </div>
+      </div>
```
we can probably move the conditional inside the parent components

kwfk commented on components/user/save.tsx:
```diff
@@ -0,0 +1,37 @@
+      <div className={styles.text}>Password</div>
+      <TextField
+        id="password"
+        defaultValue={password}
```
don't show actual password

kwfk reviewed (changes requested):
nice work! looks good but there are areas we can clean up
might need to talk to elizabeth about the editing flow
let's also add a cancel button

nice use of components :)

bry-gavino committed:
feat: fixed comments on PR

kwfk committed:
Merge branch 'main' into bryanna/user-profile-page

kwfk commented on pages/users/settings.tsx:
```diff
@@ -0,0 +1,75 @@
+      >
+        Edit
+      </Button>
+    ) : undefined;
```
let's use `null` instead of undefined

kwfk commented on pages/users/settings.tsx:
```diff
@@ -0,0 +1,75 @@
+    );
+
+  const saveButton: typeof Button =
+    setting === 0 ? undefined : (
```
null here too

kwfk commented on pages/users/settings.tsx:
```diff
@@ -0,0 +1,75 @@
+const UserProfSettings: React.FC = () => {
+  const [setting, setSetting] = useState(0);
+  const hiddenPassword = '******';
+  const editButton: typeof Button =
```
you don't have to type the variable bc Typescript will implicitly know what it is

kwfk commented on pages/users/settings.tsx:
```diff
@@ -0,0 +1,75 @@
+      </Button>
+    ) : undefined;
+
+  const userComponent: React.ReactElement =
```
no type needed here too

kwfk commented on pages/users/settings.tsx:
```diff
@@ -0,0 +1,75 @@
+      />
+    );
+
+  const saveButton: typeof Button =
```
no type needed

kwfk commented on components/user/Save/index.tsx:
```diff
@@ -0,0 +1,37 @@
+};
+
+const SettingsSave: React.FunctionComponent<Props> = ({
+  userType,
```
remove unused prop

kwfk commented on components/user/Edit/index.tsx:
```diff
@@ -0,0 +1,26 @@
+};
+
+const SettingsEdit: React.FunctionComponent<Props> = ({
+  userType,
```
remove unused prop

kwfk commented on styles/users/SettingsEdit.module.css:
```diff
@@ -0,0 +1,55 @@
+.content {
```
rename file to `Settings.module.css`

kwfk commented on components/user/Edit/Edit.module.css:
```diff
@@ -0,0 +1,14 @@
+.userInfo {
+  display: flex;
+  justify-content: space-between;
+  margin-top: 10px;
```
add
```css
height: 40px;
align-items: center;
```

kwfk replied:
this should be in the shared stylesheet

kwfk commented on components/user/Edit/Edit.module.css:
```diff
@@ -0,0 +1,14 @@
+  margin-top: 10px;
+}
+
+.text {
```
remove this style

kwfk commented on styles/users/SettingsEdit.module.css:
```diff
@@ -0,0 +1,55 @@
+}
+
+.box {
+  width: 500px;
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  margin-top: 10%;
```
change width to 60% and remove margin-top

kwfk commented on styles/users/SettingsEdit.module.css:
```diff
@@ -0,0 +1,55 @@
+.content {
+  display: flex;
+  align-items: center;
+  flex-direction: column;
```
remove flex-direction

kwfk commented on styles/users/SettingsEdit.module.css:
```diff
@@ -0,0 +1,55 @@
+  flex-direction: column;
+  justify-content: center;
+  width: 100%;
+  height: 100%;
```
change height to `calc(100vh - 85px);`

kwfk reviewed (changes requested):
I think the component naming is a little confusing. let's use `SettingsEdit` for when the user can edit the settings then `SettingsShow` as the viewing state

Also, move the component stylesheets to a shared stylesheet in `components/user/User.module.css` since it's exactly the same. Any comments added to the component stylesheet should also go in the same shared stylesheet

bry-gavino committed:
feat: fixed second round of PR comments

bry-gavino committed:
feat: fixing up PR and working on new design

has a lot of verification errors for now

bry-gavino committed:
feat: fixed up PR, git pulled, working on design

many verification errors for now

bry-gavino committed:
feat: fixed PR 2

verification errors for now

bry-gavino committed:
feat(fixing pr 2): fixing pr 2

bry-gavino committed:
feat: fixed up PR 2

bry-gavino committed:
feat: added new components to fit figma design

bry-gavino commented:
Hey Fred!! I found that it was actually easier to not use components for the "change email" part. I also ended up making a new page for the "change password" part. As always,  let me know if you want me to change anything!
<img width="1680" alt="Screen Shot 2020-11-14 at 4 21 27 PM" src="https://user-images.githubusercontent.com/60284767/99160204-72095c80-2699-11eb-869d-f27a0f4c70bf.png">
<img width="1675" alt="Screen Shot 2020-11-14 at 4 21 47 PM" src="https://user-images.githubusercontent.com/60284767/99160206-75044d00-2699-11eb-9455-9e03f53959d5.png">
<img width="1676" alt="Screen Shot 2020-11-14 at 4 21 55 PM" src="https://user-images.githubusercontent.com/60284767/99160208-76ce1080-2699-11eb-8530-3fc2ab6aaafc.png">

kwfk commented on components/user/EmailVerify/index.tsx:
```diff
@@ -0,0 +1,50 @@
+import styles from './EmailVerify.module.css';
+
+const EmailVerify: React.FC = () => {
+  const [open, setOpen] = React.useState(true);
```
import useState instead
```typescript
import { useState } from 'react'
```

kwfk commented on pages/users/password_change.tsx:
```diff
@@ -0,0 +1,54 @@
+import Layout from 'components/Layout';
```
you want page components to use `-` instead of `_`

kwfk replied:
let's also rename the component to `ChangePassword`

kwfk commented on pages/users/settings.tsx:
```diff
@@ -0,0 +1,95 @@
+import ProgressStepper from '../../components/user/ProgressStepper/index';
+import EmailVerify from '../../components/user/EmailVerify/index';
+
+function capitalizeFirstLetter(string: string): string {
+  return string.charAt(0).toUpperCase() + string.slice(1);
+}
```
we can use CSS `text-transform: capitalize` to capitalize text

kwfk commented on components/user/ProgressStepper/index.tsx:
```diff
@@ -0,0 +1,43 @@
+  const [activeStep, setActiveStep] = React.useState(0);
+  const steps = getSteps();
+
+  const handleNext: void = () => {
```
void should go after the parentheses

kwfk reviewed (commented):
let's change the stepper to be highlighted at one step at a time
i.e. https://material-ui.com/components/steppers/#non-linear-alternative-label

kwfk commented on pages/users/settings.tsx:
```diff
@@ -0,0 +1,95 @@
+
+  return (
+    <Layout title="User Profile Settings">
+      <EmailVerify />
```
let's not have the email verify show for now (we may create a custom component for toasts later)

kwfk committed:
feat(layout): add bottom margin to all pages

kwfk commented on styles/users/Settings.module.css:
```diff
@@ -0,0 +1,79 @@
+  align-items: center;
+  justify-content: center;
+  width: 100%;
+  height: calc(100vh - 85px);
```
change to `min-height`

kwfk commented on pages/users/settings.tsx:
```diff
@@ -0,0 +1,95 @@
+        <div className={styles.emailButton}>
+          {hiddenPassword}
+          <Link href="../users/password_change">
+            <Button variant="outlined" color="primary" disableElevation>
```
add onClick to go to ChangePassword page
see this to see how to use NextJS router to go to pages
https://github.com/calblueprint/nbjc/blob/a51a7af389736ee893665f6727ace20bf1a205b3/components/Layout.tsx#L46

kwfk commented on pages/users/password_change.tsx:
```diff
@@ -0,0 +1,54 @@
+              <Button color="primary" variant="contained" disableElevation>
+                Save Changes
+              </Button>
+            </Link>
```
see this to see how to use router to go to another page
https://github.com/calblueprint/nbjc/blob/a51a7af389736ee893665f6727ace20bf1a205b3/components/Layout.tsx#L46

kwfk commented on pages/users/password_change.tsx:
```diff
@@ -0,0 +1,54 @@
+              <TextField
+                size="small"
+                id="outlined-basic"
+                label={sampleUserData[0].email}
```
label is an email?

kwfk commented on pages/users/password_change.tsx:
```diff
@@ -0,0 +1,54 @@
+import Layout from 'components/Layout';
+import styles from 'styles/users/PasswordChange.module.css';
+import { Button, TextField, Link } from '@material-ui/core';
+import { sampleUserData } from '../../utils/sample-data';
```
sample user data should not be used

kwfk commented on components/user/ProgressStepper/index.tsx:
```diff
@@ -0,0 +1,43 @@
+}
+
+const ProgressStepper: React.FC = () => {
+  const [activeStep, setActiveStep] = React.useState(0);
```
import useState separately

kwfk committed:
style(user profile): minor css cleanup

Made CSS imports relative. Cleaned up some CSS naming to be more readable and moved around some
styles.

kwfk commented on styles/users/Settings.module.css:
```diff
@@ -0,0 +1,81 @@
+  align-items: center;
+  justify-content: center;
+  width: 100%;
+  height: calc(100vh - 105px);
```
change to `min-height`

kwfk commented on styles/users/PasswordChange.module.css:
```diff
@@ -0,0 +1,39 @@
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  height: calc(100vh - 105px);
```
change to `min-height`

kwfk commented on pages/users/settings.tsx:
```diff
@@ -0,0 +1,104 @@
+        <div>Password</div>
+        <div className={styles.fieldRight}>
+          {hiddenPassword}
+          <Link href="../users/password_change">
+            <Button
+              variant="outlined"
+              color="primary"
+              disableElevation
+              className={styles.fieldButton}
+            >
+              Edit
+            </Button>
+          </Link>
```
use router to link
https://github.com/calblueprint/nbjc/blob/a51a7af389736ee893665f6727ace20bf1a205b3/components/Layout.tsx#L46

kwfk reviewed (changes requested):
nice work! there some more minor changes needed here and there

## Overall Changes
center the ChangePassword component with the same CSS style as the ProfileSettings page

kwfk commented:
I pushed some minor CSS refactoring so you'll want to pull before continuing

bry-gavino committed:
feat: still need to fix things

bry-gavino committed:
feat: added new stepper design

bry-gavino committed:
feat: git pulled

bry-gavino committed:
feat: fixed minor things

kwfk commented on styles/users/PasswordChange.module.css:
```diff
@@ -0,0 +1,39 @@
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  min-height: calc(100vh - 85px);
```
this should be `100vh - 105px`

kwfk commented on styles/users/Settings.module.css:
```diff
@@ -0,0 +1,85 @@
+  align-items: center;
+  justify-content: center;
+  width: 100%;
+  min-height: calc(100vh - 85px);
```
this should be `100vh - 105px`

kwfk commented on pages/users/change-password.tsx:
```diff
@@ -0,0 +1,55 @@
+            </div>
+          </div>
+          <div className={styles.save}>
+              <Button color="primary" variant="contained" onClick={() => router.push('../users/settings')} disableElevation>
```
change to `/users/settings` no dots

kwfk commented on pages/users/change-password.tsx:
```diff
@@ -0,0 +1,55 @@
+import Layout from 'components/Layout';
```
you'll need to run prettier here
you can do so with the shortcut `cmd+option+f`

kwfk commented on pages/users/settings.tsx:
```diff
@@ -0,0 +1,97 @@
+          <Button
+            variant="outlined"
+            color="primary"
+            onClick={() => router.push('../users/change-password')}
```
change to `/users/change-password` no dots

kwfk reviewed (changes requested):
Looking good! Mentioned that the stepper is currently overflowing. Other than that, it should be nearing approval!

bry-gavino committed:
feat: override style fix bar

kwfk commented on components/user/ProgressStepper/ProgressStepper.module.css:
```diff
@@ -0,0 +1,54 @@
+
+.progress {
+  width: 100%;
+  height: 15px;
```
make height 12px to match designs

kwfk commented on components/user/ProgressStepper/ProgressStepper.module.css:
```diff
@@ -0,0 +1,54 @@
+  color: rgb(143, 143, 143);
+}
+
+.progress {
```
add `border-radius: 36px`

kwfk reviewed (approved):
one small styling change then should be good!

bry-gavino committed:
feat: ready to merge
